exampleShared/java/net/gini/android/vision/example/RuntimePermissionHandler.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    public static Builder forActivity(final Activity activity) {
+        return new Builder().forActivity(activity);
+    }
+
+    public void requestCameraPermission(final Listener listener) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            listener.permissionGranted();
+            return;
+        }
+        Dexter.withActivity(mActivity)
+                .withPermission(Manifest.permission.CAMERA)
+                .withListener(new PermissionListener() {
+                    @Override
+                    public void onPermissionGranted(final PermissionGrantedResponse response) {
+                        listener.permissionGranted();
+                    }
+
+                    @Override
+                    public void onPermissionDenied(final PermissionDeniedResponse response) {
+                        showCameraPermissionDeniedDialog(listener);
+                    }
+
+                    @Override
+                    public void onPermissionRationaleShouldBeShown(
+                            final PermissionRequest permission,
+                            final PermissionToken token) {
+                        showCameraPermissionRationale(token);
+                    }
+                })
+                .withErrorListener(new PermissionRequestErrorListener() {
+                    @Override
+                    public void onError(final DexterError error) {
+                        LOG.error("Permission error: {}", error.name());
+                    }
+                })
+                .check();
+    }
+
+    private void showCameraPermissionDeniedDialog(final Listener listener) {
+        final AlertDialog alertDialog = new AlertDialog.Builder(mActivity)
+                .setMessage(mCameraPermissionDeniedMessage)
+                .setPositiveButton(mGrantAccessButtonTitle, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(final DialogInterface dialog, final int which) {
+                        showAppDetailsSettingsScreen();
+                    }
+                })
+                .setNegativeButton(mCancelButtonTitle, null)
+                .setOnCancelListener(new DialogInterface.OnCancelListener() {
+                    @Override
+                    public void onCancel(final DialogInterface dialog) {
+                        listener.permissionDenied();
+                    }
+                })
+                .create();
+        alertDialog.show();
+    }
+
+    private void showAppDetailsSettingsScreen() {
+        final Intent intent = new Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
+        final Uri uri = Uri.fromParts("package", mActivity.getPackageName(), null);
+        intent.setData(uri);
+        mActivity.startActivity(intent);
+    }
+
+    private void showCameraPermissionRationale(final PermissionToken token) {
+        final android.support.v7.app.AlertDialog
```

Comment:
Why here the `AlertDialog` is from the support library but some in the other method it isn't?

Reply:
Auto-import... saves time, but soooo easy to import something unintended....